esy/bin/esyCommand.ml

```diff
@@ -1458,6 +1458,46 @@ let importDependencies (copts : CommonOptions.t) fromPath () =
     ~f:importBuild
     (Solution.allDependenciesBFS (Solution.root solution) solution)
 
+let show (copts : CommonOptions.t) _asJson req () =
+  let open EsyInstall in
+  let open RunAsync.Syntax in
+  let%bind (req : Req.t) = RunAsync.ofStringError @@ Req.parse req in
+  let%bind resolver = Resolver.make ~cfg:copts.cfg.installCfg ~root:copts.spec.path () in
+  let%bind resolutions =
+    RunAsync.contextf (
+      Resolver.resolve ~name:req.name ~spec:req.spec resolver
+    ) "resolving %a" Req.pp req
+  in
+  match req.spec with
+  | VersionSpec.Npm [[SemverVersion.Constraint.ANY]]
+  | VersionSpec.Opam [[OpamPackageVersion.Constraint.ANY]] ->
+    let f (res : Package.Resolution.t) = match res.resolution with
+    | Version v -> `String (Version.showSimple v)
+    | _ -> raise (Failure "unreachable")
+    in
+    `Assoc ["name", `String req.name; "versions", `List (List.map ~f resolutions)]
+    |> Yojson.Safe.pretty_to_string
+    |> print_endline;
+    return ()
+  | _ ->
+    match resolutions with
+    | [] -> errorf "No package found for %a" Req.pp req
+    | resolution::_ ->
+      let%bind pkg = RunAsync.contextf (
+          Resolver.package ~resolution resolver
+        ) "resolving metadata %a" Package.Resolution.pp resolution
+      in
+      let%bind (pkg : Package.t) = RunAsync.ofStringError pkg in
```

Comment:
As we acquired `Package.t` we are done — there's no need to use `DistResolver` (it's actually vice versa — `Resolver` uses `DistResolver`).